spec/jobs/risc_delivery_job_spec.rb

```diff
@@ -91,25 +105,44 @@
       end
     end
 
-    context 'Errno::ECONNREFUSED error' do
+    it_behaves_like 'retries and logs event when attempts exhausted', Faraday::SSLError
+    it_behaves_like 'retries and logs event when attempts exhausted', Faraday::ConnectionFailed
+
+    context 'when the job fails due to an Errno::ECONNREFUSED error' do
       before do
+        allow_any_instance_of(described_class).to receive(:analytics).and_return(job_analytics)
         # stub_request().to_raise wraps this in Faraday::ConnectionFailed, but
         # in actual usage, the original error is unwrapped
-        expect(job.faraday).to receive(:post).and_raise(Errno::ECONNREFUSED)
+        @connection = instance_double(Faraday::Connection)
+        allow(@connection).to receive(:post).and_raise(Errno::ECONNREFUSED)
+        allow(Faraday).to receive(:new).and_return(@connection)
```

Comment:
is there a way to use webmock to return the right error instead of manually stubbing `Faraday`?

Reply:
IIANM, the comments on lines 114-115 were from you and seemed to indicate that it would not simulate the real world behavior. Should I investigate further?

Reply:
LOL I had completely forgotten, you can investigate more if you want, but it's probably fine as is